src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -189,9 +189,8 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'os_disk_name', help='The name of the new VM OS disk.', arg_group='Storage')
     register_cli_argument(scope, 'os_type', help='Type of OS installed on a custom VHD. Do not use when specifying an URN or URN alias.', arg_group='Storage', **enum_choice_list(['windows', 'linux']))
     register_cli_argument(scope, 'storage_account', help="Only applicable when use with '--use-unmanaged-disk'. The name to use when creating a new storage account or referencing an existing one. If omitted, an appropriate storage account in the same resource group and location will be used, or a new one will be created.", arg_group='Storage')
-    register_cli_argument(scope, 'storage_caching', help='Storage caching type for the VM OS disk', arg_group='Storage', **enum_choice_list(['ReadWrite', 'ReadOnly']))
     register_cli_argument(scope, 'storage_sku', help='The sku of storage account to persist VM. By default, only Standard_LRS and Premium_LRS are allowed. Using with --use-unmanaged-disk, all are available.', arg_group='Storage', **enum_choice_list(SkuName))
-    register_cli_argument(scope, 'storage_container_name', help="Only applicable when use with '--use-unmanaged-disk'. Name of the storage container for the VM OS disk.", arg_group='Storage')
+    register_cli_argument(scope, 'storage_container_name', help="Only applicable when use with '--use-unmanaged-disk'. Name of the storage container for the VM OS disk. Default: vhds", arg_group='Storage')
```

Comment:
Because placing it here (or even as default in the register_cli_argument) results in the validator always failing because for N/A storage profiles, this has a value.  There are a few ways to handle this:

1) have the validators inspect the namespace as it was provided and before defaults were applied. This is what I have always preferred but it would require infrastructure changes and got shot down.

2) Apply the "MarkSpecifiedAction" to these parameters. This appends "__SET__" to the value when it is supplied by the user but not when it is defaulted, which permits you to distinguish between the a default value and the user specifying the same value as the default. However, this has proven to introduce bugs and is basically a hacky workaround, so I don't want to do that. (I tried the implementation, it was messy, I removed it).

3) Make these None and then apply the default in-code and in the help text. While not ideal, it's the simplest fix. This is also the official guidance if a complex type is the default (for example a default dictionary or list) so it's not totally contrary to guidance.

4) Keep the defaults and don't validate. This means that if the user specifies a value that is N/A, it will be silently ignored. We've gone to great lengths to throw errors when users specify invalid parameter combinations, so, especially for our most complex command, this didn't seem like something to let slide.

Given those 4 options, I really prefer 1, but I' okay with (and chose) 3.